src/libstd/ffi/c_str.rs

```diff
@@ -844,6 +950,21 @@ impl CStr {
     ///
     /// [`Cow`]: ../borrow/enum.Cow.html
     /// [`str`]: ../primitive.str.html
+    ///
+    /// # Examples
+    ///
+    /// Calling `to_string_lossy` on a `CStr` with valid unicode:
+    ///
+    /// ```
+    /// use std::borrow::Cow;
+    /// use std::ffi::CStr;
+    ///
+    /// let c_str = CStr::from_bytes_with_nul(b"foo\0").unwrap();
+    /// assert_eq!(c_str.to_string_lossy(), Cow::Borrowed("foo"));
+    /// ```
+    ///
+    /// Had `c_str` contained invalid unicode, the `to_string_lossy` call might
+    /// have returned `Cow::Borrowed("fo�")`.
```

Comment:
I figured it'd be undefined behavior to create a string w/ invalid unicode, as [your comment here](https://github.com/rust-lang/rust/pull/38839#discussion_r94711507). 

> Either way is definitely wouldn't return Cow::Borrowed it would return Cow::Owned.

Yep, good catch, I'll change that